src/hotspot/share/ci/ciMethodData.cpp

```diff
@@ -554,7 +554,7 @@ void ciMethodData::update_escape_info() {
     mdo->set_arg_local(_arg_local);
     mdo->set_arg_stack(_arg_stack);
     mdo->set_arg_returned(_arg_returned);
-    int arg_count = mdo->method()->size_of_parameters();
+    int arg_count = mdo->method()->number_of_parameters();
```

Comment:
The "argument size" in MethodData is in word sizes:

https://github.com/openjdk/jdk/blob/27a38d9093958ae4851bc61b8d3f0d71dc780823/src/hotspot/share/oops/methodData.cpp#L1021

So this change is wrong and it could missing some trailing slots in the MethodData.